drydock/manifest_builder/infrastructure/tutor_based_manifests.py

```diff
@@ -0,0 +1,93 @@
+        """Initialize the class based on the `manifest_options` from the reference.
+
+        Parameters
+        ----------
+        options: dict
+            Defines additional configuration options for the generations of the
+            configuration repository.
+
+            - ["output"]: Name of the directory to store the manifests.
+        """
+        self.options = options
+
+    def render(self, root: str, config: DrydockConfig) -> None:
+        """Register drydock custom templates and render a tutor env."""
+        with hooks.Contexts.APP("drydock-base").enter():
+            hooks.Filters.ENV_TEMPLATE_ROOTS.add_item(pkg_resources.resource_filename("drydock", "templates"))
+            hooks.Filters.ENV_TEMPLATE_TARGETS.add_items(
+                [
+                    ("default/base", "drydock"),
+                    ("default/extensions", "drydock"),
+                    ("default/kustomization.yml", "drydock"),
+                ],
+            )
+        tutor_env.save(root, config.get_data())
+        hooks.Filters.ENV_TEMPLATE_ROOTS.clear(context=hooks.Contexts.APP("drydock-base").name)
+        hooks.Filters.ENV_TEMPLATE_TARGETS.clear(context=hooks.Contexts.APP("drydock-base").name)
+
+    def relocate_env(self, src: str, dst: str) -> None:
+        """Moves the drydock rendered templates and tutor plugins to src.
+
+        At the moment we render a full tutor env with our templates under
+        'env/drydock/default'. The final drydock env is the one in 'env/drydock'
+        **plus** the plugin directory that has to be manually relocated to
+        'env/drydock/base/plugins'.
+
+        Parameters
+        ----------
+        src: str
+            The inital path where the full tutor env was rendered.
+        dst: str
+            The path to save the final drydock env.
+        """
+        base_dir = path_join(src, "env/drydock/default")
+        plugins_dir = path_join(src, "env/plugins")
+        shutil.move(base_dir, dst)
+        if os.path.exists(plugins_dir):
+            shutil.move(plugins_dir, path_join(dst, "base"))
+
+    def save(self, config: DrydockConfig) -> None:
+        """Creates an alternative tutor env with the base openedx installation.
+
+        The generated env consists of a Kustomize application that uses the original
+        tutor env as a base in addition to an `extensions` overlay to include additional
+        resources.
+
+        Parameters
+        ----------
+        config: DrydockConfig
+            A Tutor configuration extension.
+        """
+        tutor_root = config.get_root()
+        dst = path_join(tutor_root, self.options.get("output", "drydock-env"))
+        with tempfile.TemporaryDirectory() as tmpdir:
+            src = path_join(tmpdir)
+            self.render(src, config)
+            shutil.rmtree(path=dst, ignore_errors=True)
+            self.relocate_env(src, dst)
+
+        fmt.echo_info(f"Drydock environment generated in {dst}")
```

Comment:
is it relevant to print that the env was generated in a temp environment?
```
Environment generated in /tmp/tmp9ibnztrx/env
Drydock environment generated in /home/mgmdi/work/atlas/manifests-admiral/env
```

Reply:
It's not, but that comes from the tutor [`env.save()`](https://github.com/overhangio/tutor/blob/698da4b1adfa9509876a9bd6b09cb755f87b137e/tutor/env.py#L253-L265) function. The second line is what drydock prints.

Reply:
Right! :thinking: